packages/react-checkbox/src/components/Checkbox/renderCheckbox.tsx

```diff
@@ -1,22 +1,19 @@
 import * as React from 'react';
-import { getSlotsCompat } from '@fluentui/react-utilities';
-import { CheckboxState } from './Checkbox.types';
+import { getSlots } from '@fluentui/react-utilities';
+import { CheckboxState, CheckboxSlots } from './Checkbox.types';
 import { checkboxShorthandProps } from './useCheckbox';
 
-/**
- * Render the final JSX of Checkbox
- */
 export const renderCheckbox = (state: CheckboxState) => {
-  const { slots, slotProps } = getSlotsCompat(state, checkboxShorthandProps);
+  const { slots, slotProps } = getSlots<CheckboxSlots>(state, checkboxShorthandProps);
 
   return (
     <slots.root {...slotProps.root}>
-      {state.labelPosition === 'before' && <slots.label {...slotProps.label} />}
+      {state.labelPosition === 'before' && state.children}
       <div className={state.checkboxClassName}>
```

Comment:
(nit) maybe rename this to `containerClassName`. 

Or optionally, you could probably get away without this div if you put the size and margin on the `indicator` slot itself. And left or right align the `input` slot using `left: 0` or `right: 0` based on `labelPosition`.